apps/spruce/src/pages/task/taskTabs/TaskHistory/TestFailureSearchInput/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useMemo, useState } from "react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import TextInput from "@leafygreen-ui/text-input";
+import debounce from "lodash.debounce";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import { TaskHistoryOptions } from "../types";
+
+const { red } = palette;
+interface Props {
+  numMatchingResults: number;
+}
+export const TestFailureSearchInput: React.FC<Props> = ({
+  numMatchingResults,
+}) => {
+  const [failingTest, setFailingTest] = useQueryParam<string | undefined>(
+    TaskHistoryOptions.FailingTest,
+    "",
+  );
+
+  const [searchTerm, setSearchTerm] = useState(failingTest);
+  const updateQueryParamWithDebounce = useMemo(
+    () => debounce((str) => setFailingTest(str ? str : undefined), 150),
+    [setFailingTest],
+  );
+  useEffect(() => {
+    if (searchTerm !== failingTest) {
+      updateQueryParamWithDebounce(searchTerm);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm]);
+
+  return (
+    <Container>
+      <StyledInput
+        aria-label="Search Test Failure Input"
+        data-cy="search-test-failures-input"
+        label="Search Test Failures"
+        onChange={(e) => {
+          setSearchTerm(e.target.value);
+        }}
+        placeholder="Search failed test"
+        value={searchTerm}
+      />
+      {numMatchingResults === 0 && failingTest && (
+        <NoMatches>No results on this page</NoMatches>
+      )}
+    </Container>
+  );
+};
+
+const StyledInput = styled(TextInput)`
+  /* Account for chrome blue focus outline */
+  margin: 0 ${size.xxs};
+  flex: 1;
+`;
+
+const NoMatches = styled.div`
+  color: ${red.base};
+  font-weight: bold;
+  text-transform: uppercase;
+  margin-top: 20px;
+  margin-left: ${size.s};
+`;
```

Comment:
nit. I actually prefer this with more understated styling. The big red text makes it seem like there's an error, but it's a valid state.

```suggestion
const NoMatches = styled.div`
  margin-top: 20px;
  margin-left: ${size.s};
`;
```

Reply:
It's easy to miss the copy completely when the color isn't highlighted. Let me know if you think the text should be red. 

<img width="1724" alt="Screenshot 2025-05-07 at 4 58 06 PM" src="https://github.com/user-attachments/assets/d397a773-9b63-4869-b42a-7e0be5132594" />
<img width="1727" alt="Screenshot 2025-05-07 at 4 57 57 PM" src="https://github.com/user-attachments/assets/96f979f4-2819-4645-afaf-33d745b890cb" />